src/shogun/lib/SGStringList.cpp

```diff
@@ -93,12 +93,27 @@ SGStringList<T> SGStringList<T>::clone() const
 {
 	SGStringList<T> result(*this);
 
-	auto strings = SG_MALLOC(SGString<T>, num_strings);
+	auto strs = SG_MALLOC(SGString<T>, num_strings);
 
 	for (auto i : range(num_strings))
-		strings[i] = this->strings[i].clone();
+		strs[i] = this->strings[i].clone();
 
-	return SGStringList<T>(strings, num_strings, max_string_length);
+	return SGStringList<T>(strs, num_strings, max_string_length);
+}
+
+template <class T>
+bool SGStringList<T>::equals(const SGStringList<T>& other) const
+{
+	if (this->num_strings!=other.num_strings)
+		return false;
+
+	if (this->max_string_length!=other.max_string_length)
+		return false;
+
+	for (auto i : range(num_strings))
+		return ((SGString<T>)(this->strings[i])).equals(other.strings[i]);
```

Comment:
sematic error: this returns after having compared the first string in the list. I guess you wanna return false of one string is different...

Reply:
Oh yes!! Good point :P